crates/primitives-traits/src/serde_bincode_compat.rs

```diff
@@ -12,3 +13,82 @@ pub trait SerdeBincodeCompat: Sized + 'static {
 impl SerdeBincodeCompat for alloy_consensus::Header {
     type BincodeRepr<'a> = alloy_consensus::serde_bincode_compat::Header<'a>;
 }
+
+mod block_bincode {
```

Comment:
this I had to move because the trait is defined here and is generic over `SerdeBincodeCompat`, we could perhaps move this trait entirely to alloy